api/api.go

```diff
@@ -597,18 +608,40 @@ func (c *client) GetReferer() string {
 	return c.sessionCredentials.referer
 }
 
-func parseJSONError(r *http.Response) error {
-	jsonError := &JSONError{}
-	if err := json.NewDecoder(r.Body).Decode(jsonError); err != nil {
-		return err
-	}
+func parseJSONHTMLError(r *http.Response) error {
+	// check the content type of the response
+	contentType := r.Header.Get("Content-Type")
+	switch {
+	case strings.HasPrefix(contentType, "application/json"):
+		jsonErr := &JSONError{}
+		// decode JSON error response
+		err := json.NewDecoder(r.Body).Decode(jsonErr)
+		if err != nil {
+			return err
+		}
+		jsonErr.StatusCode = r.StatusCode
+		if jsonErr.Err[0].Message == "" {
+			jsonErr.Err[0].Message = r.Status
+		}
+		return jsonErr
+	case strings.HasPrefix(contentType, "text/html"):
+		htmlError := &HTMLError{}
+		// decode HTML error response
+		doc, err := goquery.NewDocumentFromReader(r.Body)
+		if err != nil {
+			return err
+		}
+		htmlError.StatusCode = r.StatusCode
+		htmlError.Message = doc.Find("h1").Text()
 
-	jsonError.StatusCode = r.StatusCode
-	if jsonError.Err[0].Message == "" {
-		jsonError.Err[0].Message = r.Status
+		if strings.TrimSpace(htmlError.Message) == "" {
+			htmlError.Message = doc.Find("title").Text()
```

Comment:
can you please validate "doc.Find("h1")" is not null before accessing ".Text()".